backend/server/rhnPackage.py

```diff
@@ -70,9 +73,42 @@ def get_package_path(server_id, pkg_spec, channel):
             and sc.channel_id = cp.channel_id
             and p.package_arch_id = pa.id
     """
-    h = rhnSQL.prepare(statement)
+    statement_with_checksum = """
+        select  p.id, p.path path, pe.epoch epoch
+        from
+                rhnPackageArch pa,
+                rhnChannelPackage cp,
+                rhnPackage p,
+                rhnChecksum cs,
+                rhnPackageEVR pe,
+                rhnServerChannel sc,
+                rhnPackageName pn,
+                rhnChannel c
+        where 1=1
+            and c.label = :channel
+            and pn.name = :name
+            and sc.server_id = :server_id
+            and pe.version = :ver
+            and pe.release = :rel
+            and c.id = sc.channel_id
+            and c.id = cp.channel_id
+            and pa.label = :arch
+            and pn.id = p.name_id
+            and p.id = cp.package_id
+            and p.evr_id = pe.id
+            and sc.channel_id = cp.channel_id
+            and p.package_arch_id = pa.id
+            and p.checksum_id = cs.id
+            and cs.checksum = :checksum
```

Comment:
Can't this be condensed in the previous query? Something like

`and (:checksum IS NULL OR cs.checksum = :checksum)